sentry-ruby/lib/sentry/ai/langchain.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require 'langchain'
+
+module Sentry
+  module AI
+    module Langchain
+      def self.patch_langchain_llms
+        # List of all LLM classes
+        llm_classes = [
+          ::Langchain::LLM::AI21,
+          ::Langchain::LLM::Anthropic,
+          ::Langchain::LLM::Azure,
+          ::Langchain::LLM::Cohere,
+          ::Langchain::LLM::GooglePalm,
+          ::Langchain::LLM::GoogleVertexAI,
+          ::Langchain::LLM::GoogleGemini,
+          ::Langchain::LLM::HuggingFace,
+          ::Langchain::LLM::LlamaCpp,
+          ::Langchain::LLM::OpenAI,
+          ::Langchain::LLM::Replicate
+        ]
+
+        llm_classes.each do |llm_class|
+          patch_llm_class(llm_class)
+        end
+      end
+
+      def self.patch_llm_class(llm_class)
+        llm_class.prepend(LangchainLLMPatch)
+      end
+
+      module LangchainLLMPatch
+        def chat(...)
+          wrap_with_sentry("chat_completions") { super(...) }
+        end
+
+        def complete(...)
+          wrap_with_sentry("chat_completions") { super(...) }
+        end
+
+        def embed(...)
+          wrap_with_sentry("embeddings") { super(...) }
+        end
+
+        def summarize(...)
+          wrap_with_sentry("chat_completions") { super(...) }
+        end
```

Comment:
Here and in the previous methods we shouldn't be using `...` syntax because unfortunately it's not gonna work with older Rubies that we still need to support. If the original methods have the same signature in case of all backends, I would suggest using the exact same signature and passing arguments to `super`.